File: nettest-shared/src/main/java/at/alladin/nettest/shared/berec/berec/collector/api/v1/dto/ApiRequest.java
package at.alladin.nettest.shared.berec.collector.api.v1.dto;

import com.fasterxml.jackson.annotation.JsonClassDescription;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * Object that is used as wrapper for every request.
 * 
 * @author alladin-IT GmbH (bp@alladin.at)
 *
 */
@io.swagger.annotations.ApiModel(description = "Object that is used as wrapper for every request.")
@JsonClassDescription("Object that is used as wrapper for every request.")
public class ApiRequest<T> extends ApiBase<T> {
	
	/**
	 * @see ApiRequestInfo
	 */
	@io.swagger.annotations.ApiModelProperty("Additional information that is sent by client alongside the request.")
	@JsonPropertyDescription("Additional information that is sent by client alongside the request.")
	@Expose
	@SerializedName("request_info")
	@JsonProperty("request_info")
	private ApiRequestInfo requestInfo;
	
	/**
	 * 
	 */
	public ApiRequest() {
		super();
	}
	
	/**
	 * 
	 * @param data
	 */
	public ApiRequest(T data) {
		super(data);
	}
	
	/**
	 * 
	 * @param data
	 * @param errors
	 */
	public ApiRequest(T data, ApiRequestInfo requestInfo) {
		super(data);
		
		this.requestInfo = requestInfo;
	}
	
	/**
	 * 
	 * @return
	 */
	public ApiRequestInfo getRequestInfo() {
		return requestInfo;
	}
	
	/**
	 * 
	 * @param requestInfo
	 */
	public void setRequestInfo(ApiRequestInfo requestInfo) {
		this.requestInfo = requestInfo;
	}
}

File: measurement-agent-swift/MeasurementAgentKit/Sources/DTO/Shared/ApiRequest.swift
// MeasurementAgentKit: ApiResponse.swift, created on 02.04.19
/*******************************************************************************
 * Copyright 2019 Benjamin Pucher (alladin-IT GmbH)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ******************************************************************************/

import Foundation

// Decoding of errors does not work if this extends from ApiBase.
// Because of that the data property from ApiBase was moved into here.

/// Object that is used as wrapper for every request.
class ApiRequest<T: Codable>: Codable /*: ApiBase<T>*/ {

    /// Actual data that is returned for the request/response.
    var data: T?

    /// Additional information that is sent by measurement agent alongside the request.
    var requestInfo: ApiRequestInfo?

    ///
    enum CodingKeys: String, CodingKey {
        case data
        case requestInfo = "request_info"
    }
}
